File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    stack = []
    result = 0
    
    for move in moves:
        for i in range(len(board)):
            doll = board[i][move-1]
            if doll:
                if stack and doll == stack[-1]:
                    board[i][move-1] = 0
                    stack.pop()
                    result += 2
                else:
                    stack.append(doll)
                    board[i][move-1] = 0
                break
                
    return result

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.ArrayDeque;
import java.util.Deque;

class Solution {
    static Deque<Integer> stack = new ArrayDeque<>();
    static int score = 0;

    public int solution(int[][] board, int[] moves) {
        for (int move : moves) {
            picking(board, move - 1);
        }
        return score;
    }

    public void picking(int[][] board, int col) {
        for (int row = 0; row < board.length; row++) {
            if (board[row][col] != 0) {
                if (!stack.isEmpty() && stack.peekLast() == board[row][col]) {
                    stack.pollLast();
                    score += 2;
                } else {
                    stack.offerLast(board[row][col]);
                }
                board[row][col] = 0;
                return;
            }
        }
    }
}